### ISSUE TITLE
Change in return type of getField() function in PETSc version 3.22.1 

### ISSUE THREAD

**Issue by jcgraciosa** [underworldcode/underworld3-archive#268]
Branch tested: development
PETSc version: 3.22.1
Python version: 3.12.7
Issue description:

Accessing the swarm fields using ```getField()``` no longer returns the field of interest as an array. 
Instead, it returns a tuple with the:
- first element corresponding to the field of interest as an array and, 
- second element corresponding to the dimensions (I think).

I think this change may have a lot of effects moving forward. 

To test: 

```
import underworld3 as uw
mesh = uw.meshing.UnstructuredSimplexBox(minCoords=(0, 0), maxCoords=(1, 1), cellSize = 1/8)   
```

**Comment by NengLu** [comment]
Yeah, I had a similar issue in https://github.com/underworldcode/underworld3/issues/257.

**Comment by jcgraciosa** [comment]
Apologies for the duplication! I assumed it was fixed.

**Comment by NengLu** [comment]
> Apologies for the duplication! I assumed it was fixed.

No worries, just close that one to follow on this. 
Shall we make the changes now to follow their updates? Or stick to the old version like 3.21.5, if only temporarily?

**Comment by lmoresi** [comment]
If this is purely in Petsc4py, then you should intercept the call in python with a branch on the version number. If it is in the C code that we have wrapped then look at the compatibility code we use.

**Comment by jcgraciosa** [comment]
As discussed with Julian and in line with Louis' comment, I'll create a wrapper method for handling the behaviour depending on the petsc4py version number.
